h2integrate/controllers/openloop_controllers.py

```diff
@@ -125,7 +127,209 @@ def setup_partials(self):
         self.declare_partials(
             of=f"{self.config.resource_name}_out",
             wrt=f"{self.config.resource_name}_in",
-            rows=arange(size),
-            cols=arange(size),
-            val=ones(size),  # Diagonal elements are 1.0
+            rows=np.arange(size),
+            cols=np.arange(size),
+            val=np.ones(size),  # Diagonal elements are 1.0
+        )
+
+
+@define
+class DemandOpenLoopControllerConfig(BaseConfig):
+    """
+    Configuration class for the DemandOpenLoopController.
+
+    This class defines the parameters required to configure the `DemandOpenLoopController`.
+
+    Attributes:
+        resource_name (str): Name of the resource being controlled (e.g., "hydrogen").
+        resource_units (str): Units of the resource (e.g., "kg/h").
+        max_capacity (float): Maximum storage capacity of the resource (in non-rate units,
+            e.g., "kg" if `resource_units` is "kg/h").
+        max_charge_percent (float): Maximum allowable state of charge (SOC) as a percentage
+            of `max_capacity`, represented as a decimal between 0 and 1.
+        min_charge_percent (float): Minimum allowable SOC as a percentage of `max_capacity`,
+            represented as a decimal between 0 and 1.
+        init_charge_percent (float): Initial SOC as a percentage of `max_capacity`, represented
+            as a decimal between 0 and 1.
+        max_charge_rate (float): Maximum rate at which the resource can be charged (in units
+            per time step, e.g., "kg/time step").
+        max_discharge_rate (float): Maximum rate at which the resource can be discharged (in
+            units per time step, e.g., "kg/time step").
+        charge_efficiency (float): Efficiency of charging the storage, represented as a decimal
+            between 0 and 1 (e.g., 0.9 for 90% efficiency).
+        discharge_efficiency (float): Efficiency of discharging the storage, represented as a
```

Comment:
Should we define this as charge and discharge efficiency or a roundtrip efficiency for the technology? I believe roundtrip efficiency is more common in literature. If we keep charge and discharge, we should think about how this maps to roundtrip efficiency.